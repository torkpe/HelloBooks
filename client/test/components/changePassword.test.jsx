import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';

import storeConfiguration from '../../store/store';
import connectedchangePassword, { ChangePassword } from '../../components/ChangePassword.jsx';

configure({ adapter: new Adapter() });

describe('<ChangePassword />', () => {
  const props = {
    setPassword: jest.fn(),
    clearSetPasswordState: jest.fn(),
    auth: {
      user: {
        id: 1
      }
    },
    passwordChange: {
      response: {
        message: 'cghjklhkgjf'
      },
      errors: {
        message: 'dfghjklhgfhj'
      }
    }
  };
  let event = {
    preventDefault: jest.fn(),
    target: {
      oldPassword: 'gfhjk',
      password: 'bjnkl',
      confirmPassword: 'hjkjhk'
    }
  };
  const initialState = {
    isLoading: false,
    errors: {},
    successfullySignedup: {},
  };
  const mockStore = configureMockStore();
  let mountedComponent;
  let store;
  let wrapper;
  beforeEach(() => {
    store = mockStore(storeConfiguration());
    mountedComponent = mount(<ChangePassword {...props} />);
    wrapper = mount(<Provider store={storeConfiguration()}><connectedchangePassword /></Provider>);
  });
  it('renders <ChangePassword /> component', () => {
    const tree = toJson(mountedComponent);
    expect(tree).toMatchSnapshot();
  });
  it('calls handleChange event', () => {
    sinon.spy(ChangePassword.prototype, 'onChange');
    mountedComponent.instance().onChange(event);
    expect(ChangePassword.prototype.onChange.calledOnce).toEqual(false);
  });
  it('calls submit event', () => {
    sinon.spy(ChangePassword.prototype, 'onSubmit');
    event = {
      preventDefault: jest.fn(),
      clearSetPasswordState: jest.fn()
    };
    const shallowComponent = shallow(<ChangePassword {...props} />);
    shallowComponent.instance().onSubmit(event);
    expect(ChangePassword.prototype.onSubmit.calledOnce).toEqual(true);
  });
  it('calls clearSetPasswordState', () => {
    const shallowComponent = shallow(<ChangePassword {...props} />);
    shallowComponent.instance().componentWillUnmount();
    expect(shallowComponent.instance().props.clearSetPasswordState).toHaveBeenCalled();
  });
  it('calls componentWillReceiveProps', () => {
    const shallowComponent = shallow(<ChangePassword {...props} />);
    shallowComponent.instance().componentWillReceiveProps(props);
    expect(shallowComponent.instance().state.isLoading).toEqual(false);
  });
});