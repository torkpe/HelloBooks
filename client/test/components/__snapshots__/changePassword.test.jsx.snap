// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePassword /> renders <ChangePassword /> component 1`] = `
<ChangePassword
  auth={
    Object {
      "user": Object {
        "id": 1,
      },
    }
  }
  clearSetPasswordState={[MockFunction]}
  passwordChange={
    Object {
      "errors": Object {
        "message": "dfghjklhgfhj",
      },
      "response": Object {
        "message": "cghjklhkgjf",
      },
    }
  }
  setPassword={[MockFunction]}
>
  <div
    className="mdl-grid"
  >
    <div
      className="contents"
    >
      <div
        className="card-enlarge card-wrapper mdl-card mdl-shadow--3dp"
      >
        <form
          onSubmit={[Function]}
        >
          <h5>
            Change Password
          </h5>
          <div
            className="card-content card-wrapper input-wrapper"
          >
            <input
              id="oldPassword"
              name="oldPassword"
              onChange={[Function]}
              placeholder="Old Password"
              type="password"
            />
          </div>
          <PasswordForm
            onChange={[Function]}
          >
            <div>
              <div
                className="card-content card-wrapper input-wrapper"
              >
                <input
                  id="newPassword"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </div>
              <div
                className="card-content card-wrapper input-wrapper"
              >
                <input
                  id="confirmPassword"
                  name="confirmPassword"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  type="password"
                />
              </div>
            </div>
          </PasswordForm>
          <button
            className="
      mdl-button
      mdl-js-button
      mdl-button--raised
      mdl-button--colored
    "
            disabled={false}
            id="button"
          >
            Reset Password
          </button>
        </form>
      </div>
    </div>
  </div>
</ChangePassword>
`;
